app/src/main/java/com/keylesspalace/tusky/TuskyApplication.kt

```diff
@@ -130,6 +131,14 @@ class TuskyApplication : Application(), HasAndroidInjector {
             editor.remove(PrefKeys.MEDIA_PREVIEW_ENABLED)
         }
 
+        if (oldVersion < 2023071701) {
+            // Default value for appTheme is now THEME_SYSTEM, so let’s save the old default value as preference to
+            // avoid upgrading user to see their theme change without action.
+            if (sharedPreferences.contains(PrefKeys.APP_THEME)) {
+                editor.putString(PrefKeys.APP_THEME, APP_THEME_DEFAULT_COMPAT)
+            }
```

Comment:
I think this check is backwards -- currently, this code will overwrite the user's previous preference, if they had set it.

What it should do is set the preference to the old default if there's no value, so this should be:

```suggestion
            if (!sharedPreferences.contains(PrefKeys.APP_THEME)) {
                editor.putString(PrefKeys.APP_THEME, APP_THEME_DEFAULT_COMPAT)
            }
```

Reply:
Shame on me >_< I must stop developing tired.